File: app/src/main/java/com/example/jacob/myapplication/logic/IConversationData.java
package com.example.jacob.myapplication.logic;

import java.util.Date;
import java.util.List;

/**
 * Created by jacob on 22/11/2015.
 */
public interface IConversationData {
    List<String> getParticipants();

    Date getMostTalkedDay();

    int getTotalDaysTalked();

    String getMostTalkedMonth();

    int getTotalMessages();

    int getParticipantCount(String pt);

    float getParticipantShare(String pt);

    float getWordsAvg(String pt);

    float getDailyAvg();

    float getRealDailyAvg();

    int getDayData(Date date);

    int getMonthData(String month);

    String getConversationName();
}

File: app/src/main/java/me/jacobrr/whatsappanalyzer/logic/IConversationData.kt
package me.jacobrr.whatsappanalyzer.logic

import java.time.LocalDate
import java.util.*

/**
 * Created by jacob on 22/11/2015.
 */
interface IConversationData {
    val participants: List<String>

    val mostTalkedDay: LocalDate

    val totalDaysTalked: Int

    val mostTalkedMonth: String

    val totalMessages: Int

    val dailyAvg: Float

    val realDailyAvg: Float

    val conversationName: String

    fun getParticipantCount(pt: String): Int

    fun getParticipantShare(pt: String): Float

    fun getWordsAvg(pt: String): Float

    fun getDayData(date: LocalDate): Int

    fun getMonthData(month: String): Int
}
